src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/EditSavesModal.tsx

```diff
@@ -0,0 +1,153 @@
+import {
+  Button,
+  Flex,
+  LabeledInput,
+  ModalDialog,
+  Spacer,
+  Text,
+  useToasts,
+} from "@artsy/palette"
+import EditIcon from "@artsy/icons/EditIcon"
+import { useTranslation } from "react-i18next"
+import { SavesArtworks_collection$data } from "__generated__/SavesArtworks_collection.graphql"
+import { Form, Formik, FormikHelpers } from "formik"
+import * as Yup from "yup"
+import { pick } from "lodash"
+import { useUpdateCollection } from "./Mutations/useUpdateCollection"
+import createLogger from "Utils/logger"
+
+interface EditSavesModalProps {
+  collection: SavesArtworks_collection$data
+  setIsEditModalOpen: React.Dispatch<React.SetStateAction<boolean>>
+}
+
+interface FormikValues {
+  name: string
+}
+
+const MAX_NAME_LENGTH = 40
+
+const logger = createLogger("EditSavesModal")
+
+export const EditSavesModal: React.FC<EditSavesModalProps> = ({
+  collection,
+  setIsEditModalOpen,
+}) => {
+  const { t } = useTranslation()
+
+  const initialValues = pick(collection, "name")
+
+  const validationSchema = Yup.object().shape({
+    name: Yup.string()
+      .required(t("collectorSaves.editListModal.fields.name.required")) // TODO: confirm copy
+      .max(MAX_NAME_LENGTH),
+  })
+
+  const { submitMutation } = useUpdateCollection()
+
+  const { sendToast } = useToasts()
+
+  const handleSubmit = async (
+    formikValues: FormikValues,
+    formikHelpers: FormikHelpers<FormikValues>
+  ) => {
+    try {
+      await submitMutation({
+        variables: {
+          input: {
+            id: collection.internalID,
+            name: formikValues.name,
+          },
+        },
+        rejectIf: res => {
+          const result = res.updateCollection?.responseOrError
+
+          return result?.__typename === "UpdateCollectionFailure"
+            ? result.mutationError
+            : false
+        },
+      })
+
+      sendToast({
+        variant: "success",
+        message: t("collectorSaves.editListModal.success"),
+      })
+    } catch (error) {
+      logger.error(error)
+      sendToast({
+        variant: "error",
+        message: error.message ?? t("common.errors.somethingWentWrong"),
+      })
+    } finally {
+      formikHelpers.setSubmitting(false)
+      setIsEditModalOpen(false)
+    }
+  }
+
+  return (
+    <Formik<FormikValues>
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+      onSubmit={handleSubmit}
+    >
+      {({
+        values,
+        errors,
+        touched,
+        handleBlur,
+        handleChange,
+        isSubmitting,
+      }) => {
+        return (
+          <ModalDialog
+            title={t("collectorSaves.editListModal.title")}
+            onClose={() => {
+              setIsEditModalOpen(false)
+            }}
+            width={["100%", 700]}
```

Comment:
I think we should open the modal window to the full screen (similar to the "create alert" modal. asked it [here](https://www.figma.com/file/wSV2KNz97Tyj8HkH2ZYAkB?node-id=466:149379#383949925)). To achieve this, we need to pass `dialogProps` and `m={0}` (like [here](https://github.com/artsy/force/blob/a6331ebbe2687a36d87a848fc585db7167f6c878/src/Apps/CollectorProfile/Routes/Saves2/Components/SelectListsForArtworkModal/SelectListsForArtworkModal.tsx#L177-L182)). Also we need to display the buttons at the bottom

| Before | After |
| ------------- | ------------- |
| <img width="376" alt="before" src="https://user-images.githubusercontent.com/3513494/222709732-ff49e74c-d482-422a-9668-605325395026.png"> | <img width="376" alt="after" src="https://user-images.githubusercontent.com/3513494/222709757-4167f8f6-388a-4ff3-8ac4-0a31b26a8094.png"> |

Reply:
Interesting, I'll wait for Ryan to chime in before making a change. 

(Currently, this works the same as the Create modal — we should change them all if this is approved.)